kod/object/passive/spell/purify.kod

```diff
@@ -79,20 +79,25 @@ messages:
    {
       local target, i, lEnchantment, bHasEnchantment, oSpell;
 
-      % If it's only self cast, spoof self as target for following code
+      % If castable on others, there can be only one target.
+      if pbCan_Cast_on_others AND Length(lTargets) = 1
+      {
+         target = First(lTargets);
+      }
+
+      % If it's only self cast, spoof self as target for the
+      % rest of the checks.
       if NOT pbCan_Cast_on_others
       {
-         lTargets = Cons(who, lTargets);
+         target = who;
       }
       
-      % Can cast spell if the 1 target item is a user
-      if Length(lTargets) <> 1
+      % If there's no target by now, something weird is going on.
+      if target = $
```

Comment:
If this is a coding error (i.e. it should "never happen"), there should be a debug print here.

Reply:
Yeah, it shouldn't happen unless something somehow is trying to cast the spell in an unconventional way.  Done.